src/librustc_mir/transform/type_check.rs

```diff
@@ -361,9 +361,9 @@ impl<'a, 'gcx, 'tcx> TypeChecker<'a, 'gcx, 'tcx> {
                         span_mirbug!(self, stmt, "bad assignment ({:?} = {:?}): {:?}",
                                      lv_ty, rv_ty, terr);
                     }
-                // FIXME: rvalue with undeterminable type - e.g. inline
-                // asm.
                 }
+                // FIXME: rvalue with undeterminable type - e.g. AggregateKind::Array branch that
+                // returns `None`.
```

Comment:
If I understand correctly, this case (the only remaining case that returns `None` now that `InlineAsm` is gone) is for empty arrays. I think that returning `[!; 0]` would be the correct behavior, removing the need to return an `Option`.